lib/Sema/TypeCheckAttr.cpp

```diff
@@ -3210,9 +3242,33 @@ SynthesizeMainFunctionRequest::evaluate(Evaluator &evaluator,
     const bool hasAsyncSupport =
         AvailabilityRange::forDeploymentTarget(context).isContainedIn(
             context.getBackDeployedConcurrencyAvailability());
-    context.Diags.diagnose(attr->getLocation(),
-                           diag::attr_MainType_without_main,
-                           nominal, hasAsyncSupport);
+
+    auto location = attr->getLocation();
+    auto fixLocation = braces.Start;
+
+    context.Diags.diagnose(location, diag::attr_MainType_without_main, nominal,
+                           hasAsyncSupport);
+
+    context.Diags.diagnose(location, diag::note_add_main_sync)
+        .fixItInsertAfter(
+            fixLocation,
+            generateMainFunctionText(context, nominal, false, false).str());
```

Comment:
```suggestion
            generateMainFunctionText(context, nominal, /* isThrows=*/false, /*isAsync=*/false).str());
```

Let’s do this for all `generateMainFunctionText` calls for clarity.